display_list/display_list_unittests.cc

```diff
@@ -1575,5 +1575,16 @@ TEST(DisplayList, SaveLayerTrueSupportsGroupOpacityWithChildSrcBlend) {
   EXPECT_TRUE(display_list->can_apply_group_opacity());
 }
 
+TEST(DisplayList, SaveLayerBoundsSnapshotsImageFilter) {
+  DisplayListBuilder builder;
+  builder.saveLayer(nullptr, true);
+  builder.drawRect({50, 50, 100, 100});
+  // This image filter should be ignored since it was not set before saveLayer
+  builder.setImageFilter(TestImageFilter1);
```

Comment:
should we test that the image filter on builder is null?

Reply:
The test isn't whether the setImageFilter() call is ignored, but whether the value it is setting is ignored when restore() does its processing. The saveLayer was supposed to snapshot this attribute when it was called so that the value of the attribute when we call restore() doesn't affect it.

Reply:
sgtm